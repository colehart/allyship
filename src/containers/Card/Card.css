.Card {
  margin: 20px 20px;
  height: 550px;
  width: 300px;
  background-color: var(--spaceship-gray);
  color: var(--deep-blue);
  display: flex;
  flex-direction: column;
  align-items: center;
  box-shadow: 0px 7px 12px 0px #00000040;
  padding: 5%;
  overflow: scroll;
  transition: 0.25s;
  border-radius: 10px;
}

.Card:hover {
  background-color: #FFFFFF;
  opacity: 1;
}

.c-title-group {
  display: flex;
  justify-content: space-between;
  align-items: flex-start;
}

.c-title {
  width: 80%;
  margin: 0;
}

.c-save {
  height: 30px;
  width: 30px;
  background-image: url('../../assets/images/sun-outline-black.svg');
  background-size: cover;
}

.c-save:hover {
  background-image: url('../../assets/images/sun-outline-color.svg');
}

.c-save.is-saved {
  background-image: url('../../assets/images/sun.svg');
}


.c-byline {
  align-self: flex-start;
  font-size: 0.75rem;
}

.c-image {
  max-height: 26%;
}

.c-link {
  color: var(--turquoise);
}

.c-link:hover {
  color: var(--denim);
}