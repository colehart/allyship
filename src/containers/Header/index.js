import React from 'react';
import { Link, NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import shuttleIcon from '../../assets/images/space-shuttle.svg';
import './Header.css';

export const Header = () => {
  return (
    <div className='Header'>
      <Link
        to='/'
        className='h-logo-group'
      >
        <img
          className='h-logo'
          src={ shuttleIcon }
          alt='A space shuttle'
        />
        <h1>Allyship</h1>
      </Link>
      <nav className='h-nav-group'>
        <NavLink
          exact to='/'
          className="h-categories btn"
        >
          Categories
        </NavLink>
        <NavLink
          exact to='/saved'
          className="h-saved btn"
        >
          Saved
          <span className='h-badge'> 0</span>
        </NavLink>
      </nav>
    </div>
  )
}

export const mapStateToProps = (state) => ({
  caughtError: state.caughtError,
  stories: state.stories
})

export default connect(mapStateToProps)(Header)