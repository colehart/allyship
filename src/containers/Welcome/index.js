import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import { Loader } from '../../components/Loader';
import { fetchStories } from '../../thunks/fetchStories';
import './Welcome.css';

export const Welcome = (props) => {
  const { isLoading } = props

  return (
    <main className='Welcome'>
      { isLoading ? <Loader /> : null }
      <section
        className={ isLoading ? 'hidden' : 'w-message'}
      >
        <h2>Welcome!</h2>
        <p>
          Allyship exists to help you find your way in the galaxy of knowledge and news any true ally must navigate.
        <br />
        <br />
          Research the latest news about a specific minority identity or issue.
        {/* , or choose your own query by entering a search keyword or phrase. */}
        </p>
      </section>
      <div
        className='w-category-group'
      >
        <Link
          to='/transgender'
          exact='true'
          className='btn'
          aria-label='Click here for the latest transgender news'
        >
          Transgender
        </Link>
        <Link
          to='/immigration'
          exact='true'
          className='btn'
          aria-label='Click here for the latest immigration news'
        >
          Immigration
        </Link>
        <Link
          to='/black-lives-matter'
          exact='true'
          aria-label='Click here for the latest black lives matter news'
          className='btn'
        >
          Black Lives Matter
        </Link>
      </div>
    </main>
  )
}

export const mapStateToProps = (state) => ({
  isLoading: state.isLoading
})

export const mapDispatchToProps = (dispatch) => ({
  fetchStories: (url) => dispatch(fetchStories(url))
})

Welcome.propTypes = {
  isLoading: PropTypes.bool.isRequired,
  fetchStories: PropTypes.func.isRequired,
}

export default connect(mapStateToProps, mapDispatchToProps)(Welcome)
