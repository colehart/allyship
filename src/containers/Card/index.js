import React from 'react';
// import { connect } from 'react-redux';
// import { PropTypes } from 'prop-types';
import './Card.css';

export const Card = (props) => {
  const {
    source,
    author,
    title,
    description,
    url,
    urlToImage,
    published,
    isSaved
  } = props

  return (
    <article className='Card'>
      <div className='c-title-group'>
        <h3 className='c-title'>{title}</h3>
        <div
          className={`c-save ${ isSaved ? 'is-saved' : '' }`}
        >
        </div>
      </div>
      <p className='c-byline'>{published} | {source} {`${author ? `| ${author}` : ''}`}</p>
      <img className='c-image' src={urlToImage} alt='provided illustration'/>
      <p className='c-description'>{description}</p>
      <a
        className='c-link'
        href={url}
        target='_blank'
        rel='noopener noreferrer'
      >
        Read article from source
      </a>
    </article>
  )
}

// export const mapDispatchToProps = (dispatch) => ({
//   toggleSave: () => dispatch(toggleSave)
// })

// Card.propTypes = {
//   toggleSaved: PropTypes.func
// }

// export default connect(null, mapDispatchToProps)(Card)