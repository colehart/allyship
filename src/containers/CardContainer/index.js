import React from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import { Loader } from '../../components/Loader';
import { Card } from '../Card';
import './CardContainer.css';

export const CardContainer = (props) => {
  const { isLoading } = props
  const cards = props.stories.map(story => {
    return <Card {...story} key={story.url} />
  })

  return (
    <main>
      <h2 className='cc-title'>Latest News</h2>
      <div className='CardContainer'>
        { isLoading ? <Loader /> : cards }
      </div>
    </main>
  )
}

export const mapStateToProps = (state) => ({
  isLoading: state.isLoading,
  stories: state.stories,
})

CardContainer.propTypes = {
  isLoading: PropTypes.bool,
  stories: PropTypes.array,
}

export default connect(mapStateToProps)(CardContainer)