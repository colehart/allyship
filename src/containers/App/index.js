import React, { Component } from 'react';
import { Route, Switch } from 'react-router-dom';
import { connect } from 'react-redux';
import { Header } from '../Header';
import { Welcome } from '../Welcome';
import { CardContainer } from '../CardContainer';
import './App.css';

export const App = (props) => {
  const { isLoading } = props

  const noMatch = ({ location }) => (
    <div className='a-404'>
      <h2>404 - Your ship has gone off course.</h2>
      <p>No routes match {location.pathname}.</p>
      <p>Please click the orange buttons in this app to navigate to your desired destination.</p>
    </div>
  )

  return (
    <div className='App'>
      <Header />
      <Switch>
        <Route
          exact path='/'
          component={ Welcome }
        />
        <Route
          exact path='/saved'
          render={() => CardContainer }
        />
        <Route
          exact path='/transgender'
          render={() => CardContainer }
        />
        <Route
          exact path='/immigration'
          render={() => CardContainer }
        />
        <Route
          exact path='/black-lives-matter'
          render={() => CardContainer }
        />
        <Route
          component={noMatch}
        />
      </Switch>
      {/* <Route
        exact path='/search'
        render={() => CardContainer }
      /> */}
    </div>
  );
}

export const mapStateToProps = (state) => ({
  isLoading: state.isLoading,
})

export default connect(mapStateToProps)(App);
