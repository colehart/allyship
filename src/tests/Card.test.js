import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { Card, mapStateToProps } from '../containers/Card';
import { mockDefaultState, mockFullState, mockStories } from './testMocks';

describe('Card', () => {
  describe('Card Component', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<Card { ...mockStories[0] } />)
    })

    it('matches the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    })
  })

  // describe('mapDispatchToProps', () => {
  //   it('should call dispatch with the toggleSaved action', () => {
  //     const mockDispatch = jest.fn()

  //     const mappedProps = mapDispatchToProps(mockDispatch)
  //     mappedProps.toggleSaved()

  //     expect(mockDispatch).toHaveBeenCalled()
  //   })
  // })
})