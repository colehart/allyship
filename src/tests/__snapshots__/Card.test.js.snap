// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Card Component matches the snapshot 1`] = `
<article
  className="Card"
>
  <div
    className="c-title-group"
  >
    <h3
      className="c-title"
    >
      What Being Transgender Looks Like, According to Stock Photography
    </h3>
    <div
      className="c-save "
    />
  </div>
  <p
    className="c-byline"
  >
    2018-10-25
     | 
    The New York Times
     
    | CLAIRE CAIN MILLER
  </p>
  <img
    alt="provided illustration"
    className="c-image"
    src="https://static01.nyt.com/images/2018/10/23/upshot/oakImage-1540334490988/oakImage-1540334490988-facebookJumbo.jpg"
  />
  <p
    className="c-description"
  >
    The most used photo of a transgender person from Getty Images’s stock photo library shows the transgender pride symbol — but only part of the person’s face.
  </p>
  <a
    className="c-link"
    href="https://www.nytimes.com/2018/10/25/upshot/what-being-transgender-looks-like-according-to-stock-photography.html"
    rel="noopener noreferrer"
    target="_blank"
  >
    Read article from source
  </a>
</article>
`;
